ldcpy/util.py

```diff
@@ -409,14 +417,42 @@ def subset_data(
         if vertical_dim_name in ds_subset.dims:
             ds_subset = ds_subset.isel({vertical_dim_name: lev})
 
-    if lat is not None:
-        ds_subset = ds_subset.sel(**{lat_coord_name: lat, 'method': 'nearest'})
-        ds_subset = ds_subset.expand_dims(lat_coord_name)
+    if latdim == 1:
+
+        if lat is not None:
+            ds_subset = ds_subset.sel(**{lat_coord_name: lat, 'method': 'nearest'})
+            ds_subset = ds_subset.expand_dims(lat_dim_name)
+
+        if lon is not None:
+            ds_subset = ds_subset.sel(**{lon_coord_name: lon + 180, 'method': 'nearest'})
+            ds_subset = ds_subset.expand_dims(lon_dim_name)
+
+    elif latdim == 2:
+
+        if lat is not None:
+            if lon is not None:
+
+                # lat is -90 to 90
+                # lon should be 0- 360
+                ad_lon = lon
+                if ad_lon < 0:
+                    ad_lon = ad_lon + 360
+
+                mlat = ds_subset[lat_coord_name].compute()
+                mlon = ds_subset[lon_coord_name].compute()
+                # euclidean dist for now....
+                di = np.sqrt(np.square(ad_lon - mlon) + np.square(lat - mlat))
+                index = np.where(di == np.min(di))
```

Comment:
```suggestion
                index = np.argmin(di)
```

Reply:
good to know - thank you!

Reply:
So I tried this suggestion, but don't understand the error.  Using the simple example from below again:
```
import numpy as np
import pandas as pd
import xarray as xr
mylat = np.array([[1., 2., 3., 4.], [1.1, 2.1, 3.1, 3.9], [1, 1.9, 2.9, 3.9]])
mylon = np.array([[3., 4., 5., 6.], [3.1, 4.1, 5.1, 5.9], [3, 3.9, 4.9, 5.9]])
mytimes = pd.date_range('2000-01-01', periods=5)
pop_data = np.random.uniform(low=0.0, high=100.0, size=(3,4,5))
ds_subset = xr.DataArray(
    pop_data, 
     coords={
        "time": mytimes,
        "TLAT": (("nlat", "nlon"), mylat, {'standard_name': 'latitude', 'units': 'degrees_north'}),
        "TLON": (("nlat", "nlon"), mylon, {'standard_name': 'longitude', 'units': 'degrees_east'})
    },
    dims=['nlat', 'nlon', 'time'] , attrs={'long_name': 'Surface Potential'}
)
```
Then I do this calculation:
```
mlat = ds_subset["TLAT"].compute()
mlon = ds_subset["TLON"].compute()
ad_lat = 3.1
ad_lon = 4
di = np.sqrt(np.square(ad_lon - mlon) + np.square(ad_lat - mlat))
index = np.argmin(di)
```
Then I get this error:
"ValueError: dimensions ('nlat', 'nlon') must have the same length as the number of data dimensions, ndim=0 "
(doing 'index = np.where(di == np.min(di))' does not give an error)